### PR TITLE
feat(anta): Added test case to verify maximum total number of routes configured and optionally verifies the warning limit

### LINKED ISSUE #784: Add test case to verify maximum total number of routes configured and optionally verifies the warning limit.
**Sample output:**

 Prefix Statistics:
                                   Sent      Rcvd     Best Paths     Best ECMP Paths
    IPv4 Unicast:                     2         1              1                   0
    IPv6 Unicast:                     0         0              0                   0
  `Configured maximum total number of routes is  $BGP_MAX_LIMIT, warning limit is  $BGP_WARN_LIMIT`
  Inbound updates dropped by reason:
    AS path loop detection: 0
    Enforced First AS: 0
    Malformed MPBGP routes: 0
    Originator ID matches local router ID: 0
    Nexthop matches local IP address: 0
    Unexpected IPv6 nexthop for IPv4 routes: 0

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Verifies maximum routes and maximum routes warning limit for the provided BGP IPv4 peer(s).

    Expected Results
    ----------------
    * Success: The test will pass if the BGP peer's maximum routes and maximum routes warning limit are equal to given limits.
    * Failure: The test will fail if the BGP peer's maximum routes and maximum routes warning limit are not equal to given limits or
    peer is not configured.

Fixes #784 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/routing/bgp.py [hunk 1]
```diff
@@ -1404,3 +1404,89 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following BGP peers are not configured or have non-zero update error counters:\n{failures}")
+
+
+class VerifyBGPPeerRouteLimit(AntaTest):
+    """Verifies maximum routes and maximum routes warning limit for the provided BGP IPv4 peer(s).
```
Comment (MaheshGSLAB) [inline anta/tests/routing/bgp.py:1410]: warning limit verification is optional so please update the doc string and success/failure accordingly

File: tests/units/anta_tests/routing/test_bgp.py [hunk 2]
```diff
@@ -4357,4 +4358,161 @@
...
+        "expected": {
+            "result": "failure",
+            "messages": [
+                "The following BGP peer(s) are not configured or maximum routes and maximum routes warning limit is not correct:\n"
+                "{'10.100.0.8': {'default': 'Not configured'}, '10.100.0.9': {'MGMT': 'Not configured'}}"
+            ],
+        },
+    },
+    {
+        "name": "failure-incorrect-conf",
```
Comment (MaheshGSLAB) [inline tests/units/anta_tests/routing/test_bgp.py:4430]:
```suggestion
        "name": "failure-incorrect-max-routes",
```

File: anta/tests/routing/bgp.py [hunk 3]
```diff
@@ -1404,3 +1404,89 @@ def test(self) -> None:
...
+        """List of BGP peers"""
+
+        class BgpPeer(BaseModel):
+            """Model for a BGP peer."""
+
+            peer_address: IPv4Address
+            """IPv4 address of a BGP peer."""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            maximum_routes: int
```
Comment (MaheshGSLAB) [inline anta/tests/routing/bgp.py:1450]: can you check if we have limit for max route and warning route to restrict the input

**Review by MaheshGSLAB — COMMENTED** [review]
First round of review

File: anta/tests/routing/bgp.py (hunk 1 above)
Comment (vitthalmagadum) [reply anta/tests/routing/bgp.py]: Updated docstring and pass/fail criteria.

File: tests/units/anta_tests/routing/test_bgp.py (hunk 2 above)
Comment (vitthalmagadum) [reply tests/units/anta_tests/routing/test_bgp.py]: Updated

File: anta/tests/routing/bgp.py (hunk 3 above)
Comment (vitthalmagadum) [reply anta/tests/routing/bgp.py:1450]: Updated input model docstring. Thanks!!

**Review by MaheshGSLAB — APPROVED** [review]
LGTM

File: anta/tests/routing/bgp.py (hunk 3 above)
Comment (carl-baillargeon) [reply anta/tests/routing/bgp.py:1450]: Please use Pydantic Field instead to restrict the input. The docstring would be simplified after.

**Review by carl-baillargeon — APPROVED** [review]
LGTM. Tested locally.
